data_pipeline_api/java_api/src/main/java/uk/ramp/access/AccessLog.java

```diff
@@ -0,0 +1,31 @@
+package uk.ramp.access;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
```

Comment:
probably a stupid question, but I'm curious, what's the relationship between YAML and JSON in this API, my speculation is that the yaml segments are being split into json like things that the json deserializer can then make into objects?

Reply:
This is a perfectly sensible question to ask. These Jackson core annotation names are a bit misleading. Even though these annotations have 'Json' their name, it will work. The Jackson ObjectMapper can be configured to parse all sorts of different formats including Yaml. In our case we are using the YAMLMapper (which is a configuration of ObjectMapper for Yaml). Given all of this, for yaml mappings you would still use the core annotations which have 'Json' in their names.